src/Features/CSharp/Portable/InitializeParameter/CSharpAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -92,5 +93,22 @@ protected override StatementSyntax CreateParameterCheckIfStatement(DocumentOptio
                 statement: ifTrueStatement,
                 @else: null);
         }
+
+        protected override Document? TryAddNullCheckToParameterDeclaration(Document document, ParameterSyntax parameterSyntax, CancellationToken cancellationToken)
+        {
+            var tree = parameterSyntax.SyntaxTree;
+            var options = (CSharpParseOptions)tree.Options;
+            if (options.LanguageVersion < LanguageVersionExtensions.CSharpNext)
+            {
+                return null;
+            }
```

Comment:
I was a bit surprised to see this. My expectation is that we would let the operation proceed without syntax checks and let later diagnostics catch that. Is there any other place where we have syntax based ops that depend on `/langVersion`?

Reply:
this is very normal on the IDE side.  remember that the user is just saying they want to add a "null check", not that they want to add `!!`.  The IDE normally takes a stance with these types of operations on choosing the most idiomatic form that is legal for the version of the language the user is on.

That way users on C#10 will get the best option they can use, wihle those on 11 will get `!!`.  

If users explicitly write out the C#11 version, then our 'upgrade project' analyzer kicks in and will help them upgrade if they want.